### PR TITLE
os: Add a default shell to mini-os

### PR DESCRIPTION
**Pull request by jcvenegas**
Need to provide a shell for vm debuging.

Fixes: #36

Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>

### CODE REVIEW

File: scripts/osbuilder.sh [hunk 1]
```diff
@@ -116,7 +116,7 @@ build_rootfs()
 	fi
 	DNF="$DNF --config=$DNF_CONF -y --installroot=${ROOTFS_DIR} --noplugins"
 	$DNF install systemd hyperstart cc-oci-runtime-extras coreutils-bin \
-		systemd-bootchart iptables-bin clear-containers-agent ${EXTRA_PKGS}
+		systemd-bootchart iptables-bin clear-containers-agent bash-bin ${EXTRA_PKGS}
```
Comment (devimc) [inline scripts/osbuilder.sh:119]: does the agent need bash ?

Comment (jcvenegas) [reply scripts/osbuilder.sh:119]: hey @devimc, it does not. It a request to allow to debug the vm.

Comment (gorozco1) [reply scripts/osbuilder.sh:119]: my vote that if this is only for debug the developer build their own image (with this tool) with bash included, and we do not need to ship with bash for security risks.

Documentation will fix that IMHO

Comment (mcastelino) [reply scripts/osbuilder.sh:119]: @devimc @gorozco1 @jodh-intel 

How is the security risk different between us including as shell and the user of the container who normally already has access to a shell as part of the container workload. If there is a material difference between the two then it makes a difference. If not we are just sacrificing usability.

So the question is, how is this shell different from the shell available in the workload. Both are root.

**Comment by jodh-intel** [comment]
Since this tool creates custom images for development/debugging, I think giving the ability to add a shell is reasonable in most scenarios. However, this might not be required for some users and does bloat the image beyond the current minimal config.

As such, why not add an option / env var to also have a shell included? Alternatively, we could just document in `make help` that if you want a shell, set `EXTRA_PKGS="bash-bin"`?

File: scripts/osbuilder.sh (hunk 1 above)
Comment (gorozco1) [reply scripts/osbuilder.sh:119]: @mcastelino well, if we provide bash this is the `vm` rootfs, the shell that is provided with the workload is in  another layer, limited within the container - imho - correct me if I am wrong.

**Comment by jodh-intel** [comment]
Any update on this?

**Comment by mcastelino** [comment]
@jcvenegas @jodh-intel @sameo @grahamwhaley  can we close on this.
IMO we should either include a shell in the default distribution. Or we should include two images (a debug and a release image). I am ok with either approach. But asking the user to build the image just to debug is not the right decision.

**Pull request closed without merging**